File: C#/Mit Polymorphismus/Polymorphismus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
#pragma warning disable IDE0028

class Polymorphismus
{
    static void Main(string[] args)
    {
        Haustier h1 = new Hund("Bello");
        Haustier h2 = new Katze("Mietze");
        Haustier h3 = new Wellensittich("Hansi");
        Haustier h4 = new Hund("Lopi");

        List<Haustier> haustiere = 
            new List<Haustier>() { h1, h2, h3, h4 };

        foreach (Haustier tier in haustiere)
        {
            tier.SchreibeName();
            tier.Fressen();
            tier.Freuen();
            tier.LautGeben();
            System.Console.WriteLine();
        }

        Console.ReadKey();
    }
}

File: TypeScript/Mit Polymorphismus/Polymorphismus.ts
import {Haustier} from "./Haustier";
import {Hund} from "./Hund";
import {Katze} from "./Katze";
import {Wellensittich} from "./Wellensittich";

class Polymorphismus{
    static hauptprogramm(){
        let h1 : Haustier = new Hund("Bello");
        let h2 : Haustier = new Katze("Mietze");
        let h3 : Haustier = new Wellensittich("Hansi");
        let h4 : Haustier = new Hund("Lopi");

        let haustiere : Haustier[] = [h1, h2, h3, h4];

        for (let tier of haustiere){
            tier.schreibeName();
            tier.fressen();
            tier.freuen();
            tier.lautGeben();
            console.log("--------------");
        }
    }

    func() {

        x : Haustier; 
        
        

        
        




    }
    
    



}

Polymorphismus.hauptprogramm();